unicodetools/src/main/java/org/unicode/text/UCD/GenerateConfusables.java

```diff
@@ -2493,4 +2493,20 @@ public String transform(Linkage<String, String> source) {
 
     };
 
+    // Copied from ICU CollectionUtilities.
+    /**
+     * Retain matching items
+     * @param <T>
+     * @param <U>
+     * @param c
+     * @param f
+     * @return
+     */
+    static <T, U extends Collection<T>> U retainAll(U c, Predicate<T> f) {
+        for (Iterator<T> it = c.iterator(); it.hasNext();) {
```

Comment:
fyi: the Java 8 way of [filtering a collection](https://www.baeldung.com/java-stream-filter-lambda#1-filtering-collections) might look like:

```
return c.stream().filter(f).collect(Collectors.toList());
```
Just an FYI, since this is probably focused on the higher-level goal of refactoring code across projects.

Reply:
Thanks. My goal was to reduce the yellow warning markers in Eclipse, not to generally modernize the code. There is a ton of code in this project using very old-fashioned Java mechanisms.